### PR TITLE
Add pubsub io

### PR DESCRIPTION
**Pull request by MartinSahlen**
This pull request adds PubsubIO source and sinks to datasplash. As the datasplash `apply-transform` handles wrapping a PCollection elegantly and is well tested, there is no tests. Also due to the existing functionality, the commit is very small, as it took very little to get PubsubIO working. It is difficult to test this locally, as it Pubsub streaming jobs must be run on the Dataflow Service. I have added a (commented out) working example with the corresponding image from the Dataflow Dashboard.
screen shot 2016-08-25 at 17 59 22

### CODE REVIEW

File: src/datasplash/pubsub.clj [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+;       (write-to-pubsub write-transformed-topic {:name "write-forwarded-interactions-to-pubsub"})))
+;
+;(defn stream-forwarded-interactions-from-pubsub
+;  [pipeline]
+;  (->> pipeline
+;       (read-from-pubsub read-transformed-subscription {:name "read-transformed-interactions-from-pubsub"})
+;       (ds/map (fn [message]
+;                 (do
+;                   (log/info (str "Got transformed message:\n" message))
+;                   message)) {:name "log-transformed-message"})))
```
Comment (bfontaine) [inline src/datasplash/pubsub.clj]: You should probably move this under `datasplash.examples`.

**Comment by torbjornvatn** [comment]
@ngrunwald @bfontaine I've moved the example code for using pubsub as a source into `examples.clj` 
I've also merged in the devel branch to be up to date

**Comment by MartinSahlen** [comment]
@m-lce Have you gotten a chance to review this?

**Comment by torbjornvatn** [comment]
@m-lce @ngrunwald I've updated this to reflect the changes in your master.
We would really appreciate if this makes it into 0.3.2 as well as we use it in our newest app 😄

**Comment by ngrunwald** [comment]
Merged and released with version 0.4.0 :) Thanks! By the way, we have not used yet datasplash in streaming mode, I have begun a few preparatory work on wrapping it nicely (for example `with-timestamp`) but is is not complete at all, for example I'd like to have an easy way to specify windowing policy. We would be very interested by this in the near future, so feel free to contact us with use cases, experiences or even further pull requests :)

**Comment by torbjornvatn** [comment]
Thanks a bunch! I'll bump our dependencies right away 😀

We have just started using Datasplash in an app that utilizes a streaming Dataflow pipeline, so we'll come back with suggestions/pull requests if we run into anything windowing related

**Pull request merged**